tests/test_comments.py

```diff
@@ -280,3 +292,676 @@ def test_find_unread_does_not_include_deleted_comments(self):
+        payload = self._create_payload('move', self.user, source, destination, self.file._id)
+        update_file_guid_referent(self=None, node=destination['node'], event_type='addon_file_moved', payload=payload)
+        self.guid.reload()
+
+        file_node = FileNode.resolve_class(self.provider, FileNode.FILE).get_or_create(destination['node'], self._format_path('{}{}'.format(destination['path'], file_name), file_id=self.file._id))
+        assert_equal(self.guid._id, file_node.get_guid()._id)
+        file_comments = Comment.find(Q('root_target', 'eq', self.guid._id))
+        assert_equal(file_comments.count(), 1)
+
+    @unittest.skip
+    def test_comments_move_when_folder_moved_from_subfolder_to_root(self):
+        source = {
+            'path': '/subfolder2/subfolder/',
+            'node': self.project,
+            'provider': self.provider
+        }
+        destination = {
+            'path': '/subfolder/',
+            'node': self.project,
+            'provider': self.provider
+        }
+        file_name = 'file.txt'
+        self._create_file_with_comment(node=source['node'], path='{}{}'.format(source['path'], file_name))
+        payload = self._create_payload('move', self.user, source, destination, self.file._id)
+        update_file_guid_referent(self=None, node=destination['node'], event_type='addon_file_moved', payload=payload)
+        self.guid.reload()
+
+        file_node = FileNode.resolve_class(self.provider, FileNode.FILE).get_or_create(destination['node'], self._format_path('{}{}'.format(destination['path'], file_name), file_id=self.file._id))
+        assert_equal(self.guid._id, file_node.get_guid()._id)
+        file_comments = Comment.find(Q('root_target', 'eq', self.guid._id))
+        assert_equal(file_comments.count(), 1)
+
+    @unittest.skip
+    def test_comments_move_when_folder_moved_from_project_to_component(self):
+        source = {
+            'path': '/subfolder/',
+            'node': self.project,
+            'provider': self.provider
+        }
+        destination = {
+            'path': '/subfolder/',
+            'node': self.component,
+            'provider': self.provider
+        }
+        file_name = 'file.txt'
+        self._create_file_with_comment(node=source['node'], path='{}{}'.format(source['path'], file_name))
+        payload = self._create_payload('move', self.user, source, destination, self.file._id)
+        update_file_guid_referent(self=None, node=destination['node'], event_type='addon_file_moved', payload=payload)
+        self.guid.reload()
+
+        file_node = FileNode.resolve_class(self.provider, FileNode.FILE).get_or_create(destination['node'], self._format_path('{}{}'.format(destination['path'], file_name), file_id=self.file._id))
+        assert_equal(self.guid._id, file_node.get_guid()._id)
+        file_comments = Comment.find(Q('root_target', 'eq', self.guid._id))
+        assert_equal(file_comments.count(), 1)
+
+    @unittest.skip
```

Comment:
The payload format when an addon folder is moved or renamed is not consistent between addons or even within the same addon (with the different possible move/rename actions) -- some include a "children" field which includes the folder's serialized contents and some do not. The tests still pass, but they are not quite accurate so @sloria suggested skipping them.

We can remove the skips if/when waterbutler returns a consistently (or at least predictably) formatted payload.